dtk/process.py

```diff
@@ -506,28 +518,239 @@ def nextpow2(i):
     except IndexError:
         L = data.shape[0]  # length of data if (n,)
     # calculate the closest power of 2 for the length of the data
     n = nextpow2(L)
-    Y = fft(data, n)/L  # divide by L for scaling
+
+    Y = fft(data, n, norm="forward") 
     f = fftfreq(n, d=time)
-    # f = sampleRate/2.*linspace(0, 1, n)
+    # f = sample_rate/2.*linspace(0, 1, n)
     # print 'f =', f, f.shape, type(f)
-    frequency = f[1:n//2]
+    frequency = f[int(remove_dc_component):n//2]
     try:
         # multiply by 2 because we take half the vector
-        amplitude = 2*abs(Y[:, 1:n//2]).T
+        amplitude = 2*abs(Y[:, int(remove_dc_component):n//2]).T
         # power = abs(Y[:, 1:n/2])**2
     except IndexError:
-        amplitude = 2*abs(Y[1:n//2])
+        amplitude = 2*abs(Y[int(remove_dc_component):n//2])
         # power = abs(Y[1:n/2])**2
+        
+    #Correct the dc component. It may not be multiplied by two, 
+    #the full spectrum [0,f_s[ (or equiv. ]-f_n, f_n[) only includes it once.
+    if not remove_dc_component:
+        amplitude[0] = amplitude[0] / 2
+
     return frequency, amplitude
 
 
+def pow_spectrum(data, sample_rate, remove_dc_component=False):
+    """
+    Return the power spectrum of a signal::
+    
+        S(f) = |X(f)|^2
+    
+    Notes
+    ----- 
+    - pow_spectrum() performs zero-padding. Parseval's 
+      theorem is satisfied for the padded input signal. Provide input signals
+      with 2^p samples to prevent zero-padding. 
+    - The power contributions of positive and negative frequencies are 
+      combined in the positive half spectrum so that the results satisfy
+      Parseval's theoreom on the interval [0, f_N].
+    - If the dc component is removed with remove_dc_component=True, the results
+      do not satisfy Parseval's theorem.
+
+    Parameters
+    ----------
+    data : ndarray, shape (m,) or shape(n,m)
+        The array of time signals where n is the number of variables and m is
+        the number of time steps.
+    sample_rate : int
+        The signal sampling rate in Hertz.
+    remove_dc_component : bool, optional
+        If True, the DC component (f = 0) is not included in the returned 
+        spectrum ]0,f_N[. If False the returned spectrum covers
+        [0, f_N[. The default is True. 
+    
+
+    Returns
+    -------
+    frequency : ndarray, shape (p,)
+        The frequencies where p is a power of 2 close to m.
+    power : ndarray, shape (p,n)
+        The power at each frequency.
+        
+    Examples
+    --------
+
+    Create the power spectrum of a rect pulse and plot in time and frequency
+    domain. Note how the power of frequencies f>0 is larger then f=0 because
+    the positive frequencies inlclude the contribution of negative frequencies.
+    As a result, the mean power in the displayed half spectrum equals the 
+    the mean power of the input signal. 
+
+    .. plot::
+       :context: reset
+       :source-include:
```

Comment:
```suggestion
       :include-source:
```

I wrote in wrong order before.